src/Microsoft.FeatureManagement/FeatureManager.cs

```diff
@@ -312,9 +312,17 @@ private async ValueTask<EvaluationEvent> EvaluateFeature<TContext>(string featur
                     {
                         if (targetingContext == null)
                         {
-                            Logger?.LogWarning(useContext ?
-                                $"The {nameof(TargetingContext)} required for variant assignment was not provided." :
-                                $"The {nameof(TargetingContext)} required for variant assignment was not retrieved from the {nameof(ITargetingContextAccessor)}.");
+                            string message;
+
+                            if (useContext) {
+                                message = $"A {nameof(TargetingContext)} required for variant assignment was not provided.";
+                            } else if (TargetingContextAccessor == null) {
```

Comment:
I don't believe we use `else if` on the same line as a bracket anywhere else in the code base.

Should be like

```
if
{
}
else if ()
{
}
else
{
}

```